x-pack/plugins/maps/public/store/map.js

```diff
@@ -221,8 +223,13 @@ export function map(state = INITIAL_STATE, action) {
         }
       };
     case SET_SELECTED_LAYER:
-      const match = state.layerList.find(layer => layer.id === action.selectedLayerId);
-      return { ...state, selectedLayerId: match ? action.selectedLayerId : null };
+      const selectedMatch = state.layerList.find(layer => layer.id === action.selectedLayerId);
+      return { ...state, selectedLayerId: selectedMatch ? action.selectedLayerId : null };
+    case ADD_TRANSIENT_LAYER:
+      const transientMatch = state.layerList.find(layer => layer.id === action.transientLayerId);
+      return { ...state, __transientLayerId: transientMatch ? action.transientLayerId : null };
+    case CLEAR_TRANSIENT_LAYER:
```

Comment:
Do we need this `clear_transient_layer` action at all?

It is identical to setting this to null. We have `setSelectedLayer(null)` as a precursor.